File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] res = s.trim().split("\\s+");
        int i = 0, j = res.length - 1;
        while (i < j) {
            String t = res[i];
            res[i] = res[j];
            res[j] = t;
            i++; j--;
        }
        String result = String.join(" ", res);
        return result;
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        int i = 0, n = s.size();
        vector<string> res;
        while (i < n) {
            while (s[i] == ' ') i++;
            if (i >= n) break;
            string ans = "";
            while (s[i] != ' ' && i < n) {
                ans += s[i];
                i++;
            }
            res.push_back(ans);
        }
        reverse(res.begin(), res.end());
        string fin = "";
        for (int i = 0; i < res.size(); i++) {
            fin += res[i];
            if (i < res.size() - 1) {
                fin += " ";
            }
        }
        return fin;
    }
};